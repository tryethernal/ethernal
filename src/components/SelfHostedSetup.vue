<template>
  <v-app class="setup-bg">
    <v-container class="d-flex align-center justify-center setup-container" fluid>
      <v-row align="center" justify="center" class="w-100">
        <v-col cols="12" md="7" lg="5">
          <v-card class="pa-8 setup-card" elevation="10">
            <div class="d-flex justify-center align-center mb-8">
              <h1 class="ml-3 mb-0 setup-title">Welcome! Let's Get Your Ethernal Instance Ready</h1>
            </div>
            <div class="wizard-stepper d-flex justify-space-between align-center mb-8">
              <div class="wizard-step" :class="{ active: step === 1, done: step > 1 }">
                <v-icon class="step-icon" :color="step === 1 ? 'primary' : 'grey-lighten-1'" size="22">mdi-account-plus</v-icon>
                <span class="step-label">Create Admin User</span>
              </div>
              <v-icon v-if="true" class="wizard-arrow" size="20" color="grey-lighten-1">mdi-arrow-right-thin</v-icon>
              <div class="wizard-step" :class="{ active: step === 2, done: step > 2 }">
                <v-icon class="step-icon" :color="step === 2 ? 'primary' : 'grey-lighten-1'" size="22">mdi-rocket-launch-outline</v-icon>
                <span class="step-label">Create Explorer</span>
              </div>
              <v-icon v-if="true" class="wizard-arrow" size="20" color="grey-lighten-1">mdi-arrow-right-thin</v-icon>
              <div class="wizard-step" :class="{ active: step === 3 }">
                <v-icon class="step-icon" :color="step === 3 ? 'primary' : 'grey-lighten-1'" size="22">mdi-check-circle-outline</v-icon>
                <span class="step-label">Done</span>
              </div>
            </div>

            <v-divider class="mb-7"></v-divider>

            <v-window v-model="step">
              <v-window-item :value="1">
                <SelfHostedSetupUser @user-created="onUserCreated" :loading="loading" />
              </v-window-item>
              <v-window-item :value="2">
                <SelfHostedSetupExplorer @explorer-created="onExplorerCreated" :loading="loading" />
              </v-window-item>
              <v-window-item :value="3">
                <SelfHostedSetupDone :explorer="explorer" />
              </v-window-item>
            </v-window>
          </v-card>
        </v-col>
      </v-row>
    </v-container>
  </v-app>
</template>

<script setup>
import { ref, onMounted } from 'vue';
import SelfHostedSetupUser from './SelfHostedSetupUser.vue';
import SelfHostedSetupExplorer from './SelfHostedSetupExplorer.vue';
import SelfHostedSetupDone from './SelfHostedSetupDone.vue';

const step = ref(1);
const loading = ref(false);
const explorer = ref(null);

function onUserCreated() {
  step.value = 2;
}

function onExplorerCreated(explorerData) {
  explorer.value = explorerData;
  step.value = 3;
}

onMounted(() => {
  if (localStorage.getItem('apiToken')) {
    step.value = 2;
  }
});
</script>

<style scoped>
.setup-bg {
  min-height: 100vh;
  min-width: 100vw;
  background: linear-gradient(180deg, #f8fafc 0%, #f3f4f6 100%);
}
.setup-container {
  min-height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
}
.setup-card {
  background: #fff;
  border-radius: 18px;
  box-shadow: 0 4px 32px rgba(0,0,0,0.08);
}
.setup-title {
  font-weight: 700;
  font-size: 1.4rem;
  letter-spacing: -1px;
}
.wizard-stepper {
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 0.5rem;
}
.wizard-step {
  display: flex;
  flex-direction: column;
  align-items: center;
  min-width: 90px;
  opacity: 0.5;
  transition: opacity 0.2s, color 0.2s;
}
.wizard-step.active {
  opacity: 1;
}
.wizard-step.done {
  opacity: 0.8;
}
.step-icon {
  margin-bottom: 2px;
}
.step-label {
  font-size: 1rem;
  font-weight: 500;
  margin-top: 2px;
  text-align: center;
}
.wizard-arrow {
  margin: 0 8px 0 8px;
  opacity: 0.5;
  vertical-align: middle;
}
</style> 