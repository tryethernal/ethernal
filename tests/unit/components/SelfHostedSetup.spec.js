import SelfHostedSetup from '@/components/SelfHostedSetup.vue';

const stubs = ['SelfHostedSetupUser', 'SelfHostedSetupExplorer', 'SelfHostedSetupDone'];

describe('SelfHostedSetup.vue', () => {
    it('renders the initial user step', async () => {
        const wrapper = mount(SelfHostedSetup, {
            global: {
                stubs
            }
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('moves to explorer step on user-created', async () => {
        const wrapper = mount(SelfHostedSetup, {
            global: {
                stubs
            }
        });
        // Simulate user-created event
        await wrapper.findComponent({ name: 'SelfHostedSetupUser' }).vm.$emit('user-created');
        await flushPromises();
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('moves to done step on explorer-created', async () => {
        const wrapper = mount(SelfHostedSetup, {
            global: {
                stubs
            }
        });
        // Simulate user-created event
        await wrapper.findComponent({ name: 'SelfHostedSetupUser' }).vm.$emit('user-created');
        await flushPromises();
        // Simulate explorer-created event
        await wrapper.findComponent({ name: 'SelfHostedSetupExplorer' }).vm.$emit('explorer-created', { id: 42, slug: 'slug' });
        await flushPromises();
        expect(wrapper.html()).toMatchSnapshot();
    });
}); 