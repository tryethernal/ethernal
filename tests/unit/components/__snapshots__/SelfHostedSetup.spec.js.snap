// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SelfHostedSetup.vue > moves to done step on explorer-created 1`] = `
"<div data-v-26a1cee2="" class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr setup-bg">
  <div class="v-application__wrap">
    <div data-v-26a1cee2="" class="v-container v-container--fluid v-locale--is-ltr d-flex align-center justify-center setup-container">
      <div data-v-26a1cee2="" class="v-row align-center justify-center w-100">
        <div data-v-26a1cee2="" class="v-col-md-7 v-col-lg-5 v-col-12">
          <div data-v-26a1cee2="" class="v-card v-theme--light v-card--density-default elevation-10 v-card--variant-elevated pa-8 setup-card">
            <!---->
            <div class="v-card__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                <!---->
                <div class="v-progress-linear__background"></div>
                <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__indeterminate">
                    <div class="v-progress-linear__indeterminate long"></div>
                    <div class="v-progress-linear__indeterminate short"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <!---->
            <div data-v-26a1cee2="" class="d-flex justify-center align-center mb-8">
              <h1 data-v-26a1cee2="" class="ml-3 mb-0 setup-title">Welcome! Let's Get Your Ethernal Instance Ready</h1>
            </div>
            <div data-v-26a1cee2="" class="wizard-stepper d-flex justify-space-between align-center mb-8">
              <div data-v-26a1cee2="" class="wizard-step done"><i data-v-26a1cee2="" class="mdi-account-plus mdi v-icon notranslate v-theme--light text-grey-lighten-1 step-icon" style="font-size: 22px; height: 22px; width: 22px;" aria-hidden="true"></i><span data-v-26a1cee2="" class="step-label">Create Admin User</span></div><i data-v-26a1cee2="" class="mdi-arrow-right-thin mdi v-icon notranslate v-theme--light text-grey-lighten-1 wizard-arrow" style="font-size: 20px; height: 20px; width: 20px;" aria-hidden="true"></i>
              <div data-v-26a1cee2="" class="wizard-step done"><i data-v-26a1cee2="" class="mdi-rocket-launch-outline mdi v-icon notranslate v-theme--light text-grey-lighten-1 step-icon" style="font-size: 22px; height: 22px; width: 22px;" aria-hidden="true"></i><span data-v-26a1cee2="" class="step-label">Create Explorer</span></div><i data-v-26a1cee2="" class="mdi-arrow-right-thin mdi v-icon notranslate v-theme--light text-grey-lighten-1 wizard-arrow" style="font-size: 20px; height: 20px; width: 20px;" aria-hidden="true"></i>
              <div data-v-26a1cee2="" class="wizard-step active"><i data-v-26a1cee2="" class="mdi-check-circle-outline mdi v-icon notranslate v-theme--light text-primary step-icon" style="font-size: 22px; height: 22px; width: 22px;" aria-hidden="true"></i><span data-v-26a1cee2="" class="step-label">Done</span></div>
            </div>
            <hr data-v-26a1cee2="" class="v-divider v-theme--light mb-7" aria-orientation="horizontal" role="separator">
            <div data-v-26a1cee2="" class="v-window v-theme--light">
              <div class="v-window__container">
                <transition-stub data-v-26a1cee2="" appear="false" persisted="false" css="true" disabled="true">
                  <div class="v-window-item" style="display: none;">
                    <self-hosted-setup-user-stub data-v-26a1cee2="" loading="false"></self-hosted-setup-user-stub>
                  </div>
                </transition-stub>
                <transition-stub data-v-26a1cee2="" appear="false" persisted="false" css="true" disabled="true">
                  <div class="v-window-item" style="display: none;">
                    <self-hosted-setup-explorer-stub data-v-26a1cee2="" loading="false"></self-hosted-setup-explorer-stub>
                  </div>
                </transition-stub>
                <transition-stub data-v-26a1cee2="" appear="false" persisted="false" css="true" disabled="true">
                  <div class="v-window-item v-window-item--active" style="">
                    <self-hosted-setup-done-stub data-v-26a1cee2="" explorer="[object Object]"></self-hosted-setup-done-stub>
                  </div>
                </transition-stub>
                <!---->
              </div>
              <!---->
            </div>
            <!---->
            <!----><span class="v-card__underlay"></span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`SelfHostedSetup.vue > moves to explorer step on user-created 1`] = `
"<div data-v-26a1cee2="" class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr setup-bg">
  <div class="v-application__wrap">
    <div data-v-26a1cee2="" class="v-container v-container--fluid v-locale--is-ltr d-flex align-center justify-center setup-container">
      <div data-v-26a1cee2="" class="v-row align-center justify-center w-100">
        <div data-v-26a1cee2="" class="v-col-md-7 v-col-lg-5 v-col-12">
          <div data-v-26a1cee2="" class="v-card v-theme--light v-card--density-default elevation-10 v-card--variant-elevated pa-8 setup-card">
            <!---->
            <div class="v-card__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                <!---->
                <div class="v-progress-linear__background"></div>
                <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__indeterminate">
                    <div class="v-progress-linear__indeterminate long"></div>
                    <div class="v-progress-linear__indeterminate short"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <!---->
            <div data-v-26a1cee2="" class="d-flex justify-center align-center mb-8">
              <h1 data-v-26a1cee2="" class="ml-3 mb-0 setup-title">Welcome! Let's Get Your Ethernal Instance Ready</h1>
            </div>
            <div data-v-26a1cee2="" class="wizard-stepper d-flex justify-space-between align-center mb-8">
              <div data-v-26a1cee2="" class="wizard-step done"><i data-v-26a1cee2="" class="mdi-account-plus mdi v-icon notranslate v-theme--light text-grey-lighten-1 step-icon" style="font-size: 22px; height: 22px; width: 22px;" aria-hidden="true"></i><span data-v-26a1cee2="" class="step-label">Create Admin User</span></div><i data-v-26a1cee2="" class="mdi-arrow-right-thin mdi v-icon notranslate v-theme--light text-grey-lighten-1 wizard-arrow" style="font-size: 20px; height: 20px; width: 20px;" aria-hidden="true"></i>
              <div data-v-26a1cee2="" class="wizard-step active"><i data-v-26a1cee2="" class="mdi-rocket-launch-outline mdi v-icon notranslate v-theme--light text-primary step-icon" style="font-size: 22px; height: 22px; width: 22px;" aria-hidden="true"></i><span data-v-26a1cee2="" class="step-label">Create Explorer</span></div><i data-v-26a1cee2="" class="mdi-arrow-right-thin mdi v-icon notranslate v-theme--light text-grey-lighten-1 wizard-arrow" style="font-size: 20px; height: 20px; width: 20px;" aria-hidden="true"></i>
              <div data-v-26a1cee2="" class="wizard-step"><i data-v-26a1cee2="" class="mdi-check-circle-outline mdi v-icon notranslate v-theme--light text-grey-lighten-1 step-icon" style="font-size: 22px; height: 22px; width: 22px;" aria-hidden="true"></i><span data-v-26a1cee2="" class="step-label">Done</span></div>
            </div>
            <hr data-v-26a1cee2="" class="v-divider v-theme--light mb-7" aria-orientation="horizontal" role="separator">
            <div data-v-26a1cee2="" class="v-window v-theme--light">
              <div class="v-window__container">
                <transition-stub data-v-26a1cee2="" appear="false" persisted="false" css="true" disabled="true">
                  <div class="v-window-item" style="display: none;">
                    <self-hosted-setup-user-stub data-v-26a1cee2="" loading="false"></self-hosted-setup-user-stub>
                  </div>
                </transition-stub>
                <transition-stub data-v-26a1cee2="" appear="false" persisted="false" css="true" disabled="true">
                  <div class="v-window-item v-window-item--active" style="">
                    <self-hosted-setup-explorer-stub data-v-26a1cee2="" loading="false"></self-hosted-setup-explorer-stub>
                  </div>
                </transition-stub>
                <transition-stub data-v-26a1cee2="" appear="false" persisted="false" css="true" disabled="true">
                  <div class="v-window-item" style="display: none;">
                    <!---->
                  </div>
                </transition-stub>
                <!---->
              </div>
              <!---->
            </div>
            <!---->
            <!----><span class="v-card__underlay"></span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`SelfHostedSetup.vue > renders the initial user step 1`] = `
"<div data-v-26a1cee2="" class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr setup-bg">
  <div class="v-application__wrap">
    <div data-v-26a1cee2="" class="v-container v-container--fluid v-locale--is-ltr d-flex align-center justify-center setup-container">
      <div data-v-26a1cee2="" class="v-row align-center justify-center w-100">
        <div data-v-26a1cee2="" class="v-col-md-7 v-col-lg-5 v-col-12">
          <div data-v-26a1cee2="" class="v-card v-theme--light v-card--density-default elevation-10 v-card--variant-elevated pa-8 setup-card">
            <!---->
            <div class="v-card__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                <!---->
                <div class="v-progress-linear__background"></div>
                <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__indeterminate">
                    <div class="v-progress-linear__indeterminate long"></div>
                    <div class="v-progress-linear__indeterminate short"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <!---->
            <div data-v-26a1cee2="" class="d-flex justify-center align-center mb-8">
              <h1 data-v-26a1cee2="" class="ml-3 mb-0 setup-title">Welcome! Let's Get Your Ethernal Instance Ready</h1>
            </div>
            <div data-v-26a1cee2="" class="wizard-stepper d-flex justify-space-between align-center mb-8">
              <div data-v-26a1cee2="" class="wizard-step active"><i data-v-26a1cee2="" class="mdi-account-plus mdi v-icon notranslate v-theme--light text-primary step-icon" style="font-size: 22px; height: 22px; width: 22px;" aria-hidden="true"></i><span data-v-26a1cee2="" class="step-label">Create Admin User</span></div><i data-v-26a1cee2="" class="mdi-arrow-right-thin mdi v-icon notranslate v-theme--light text-grey-lighten-1 wizard-arrow" style="font-size: 20px; height: 20px; width: 20px;" aria-hidden="true"></i>
              <div data-v-26a1cee2="" class="wizard-step"><i data-v-26a1cee2="" class="mdi-rocket-launch-outline mdi v-icon notranslate v-theme--light text-grey-lighten-1 step-icon" style="font-size: 22px; height: 22px; width: 22px;" aria-hidden="true"></i><span data-v-26a1cee2="" class="step-label">Create Explorer</span></div><i data-v-26a1cee2="" class="mdi-arrow-right-thin mdi v-icon notranslate v-theme--light text-grey-lighten-1 wizard-arrow" style="font-size: 20px; height: 20px; width: 20px;" aria-hidden="true"></i>
              <div data-v-26a1cee2="" class="wizard-step"><i data-v-26a1cee2="" class="mdi-check-circle-outline mdi v-icon notranslate v-theme--light text-grey-lighten-1 step-icon" style="font-size: 22px; height: 22px; width: 22px;" aria-hidden="true"></i><span data-v-26a1cee2="" class="step-label">Done</span></div>
            </div>
            <hr data-v-26a1cee2="" class="v-divider v-theme--light mb-7" aria-orientation="horizontal" role="separator">
            <div data-v-26a1cee2="" class="v-window v-theme--light">
              <div class="v-window__container">
                <transition-stub data-v-26a1cee2="" appear="false" persisted="false" css="true" disabled="true">
                  <div class="v-window-item v-window-item--active">
                    <self-hosted-setup-user-stub data-v-26a1cee2="" loading="false"></self-hosted-setup-user-stub>
                  </div>
                </transition-stub>
                <transition-stub data-v-26a1cee2="" appear="false" persisted="false" css="true" disabled="true">
                  <div class="v-window-item" style="display: none;">
                    <!---->
                  </div>
                </transition-stub>
                <transition-stub data-v-26a1cee2="" appear="false" persisted="false" css="true" disabled="true">
                  <div class="v-window-item" style="display: none;">
                    <!---->
                  </div>
                </transition-stub>
                <!---->
              </div>
              <!---->
            </div>
            <!---->
            <!----><span class="v-card__underlay"></span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
